public/js/user-actions.js

```diff
@@ -13,13 +16,56 @@ function UserActions(localDispatcher) {
 }
 
 
-UserActions.prototype.setCurrentUser = function(user) {
-  console.log('UserActions: setting current user', user);
-  this.dispatcher.dispatch({
-    actionType: 'set-user',
-    user: user,
-  });
-};
+UserActions.prototype = assign({}, {
+
+  signIn: function(accessToken, opt) {
+    opt = opt || {};
+    if (!opt.jquery) {
+      opt.jquery = $;
+    }
+
+    opt.jquery.ajax({
+      data: {
+        access_token: accessToken,
+      },
+      method: 'post',
+      url: '/api/auth/sign-in/',
+      context: this,
+    }).then(function(data) {
+
+      console.log('login succeeded');
+      this.setCurrentUser({
+        email: data.buyer_email,
+        is_logged_in: true,
+      });
+
+      console.log('setting CSRF token for subsequent requests:',
+                  data.csrf_token);
+      opt.jquery.ajaxSetup({
+        headers: {
+          'X-CSRFToken': data.csrf_token,
+        },
```

Comment:
I don't think this is how you want to go about it:

1. Any future calls to ajaxSetup will smother this.
2. Any attempts to set headers on individual requests will smother this.
3. Its use isn't recommended by jQuery.

Alternately, maintain an object in localstorage or memory that you use as per-request defaults? e.g.

    localStorage.setItem('csrftoken', data.csrf_token);

    // Then in each subsequent call to $.ajax
    $.ajax(assign({}, localStorage.getItem('csrftoken'), {
      method: 'post',
      url: '/api/auth/sign-in/'
    }));